stl/inc/flat_map

```diff
@@ -306,6 +306,19 @@ public:
     flat_map(sorted_unique_t _S, initializer_list<value_type> _I, const Allocator& _Alloc)
         : flat_map(_S, _I, key_compare(), _Alloc) {}
 
+    // 24.6.8.4 capacity
+    [[nodiscard]] bool empty() const noexcept {

```

Comment:
conventionally, we use `_NODISCARD` macro for this, and we can add them wherever suitable(for example, we call `size()` and `max_size()` solely for their returned value, so we can add `_NODISCARD` for them too)

Reply:
For the particular method we use `_NODISCARD_EMPTY_MEMBER` to have a more meaningful message

Reply:
Done. I added some `_NODISCARD`s to the iterator functions and elemental access functions too.